### DISCUSSION TITLE
How to connect with Azure SQL server driver 17 and straight tcp 1433

### DISCUSSION THREAD

**Discussion by trx-jay** [mkleehammer/pyodbc#1123] [Q&A] [answered]
Hi,

I have a tricky customer who can't resolve the initial connection attempt via UDP 1434, is there a way to instantiate the connection 100% by tcp on port 1433? 

There's a package /usr/local/lib/python3.8/dist-packages/sqlserverport/sqlserverport.py in my system that seems to keep forcing a UDP port 1434 connection but the cloud infrastructure my client has can't resolve this

Disclosure - I'm going linux to Azure SQL server via a strongswan VPN 

My connection string is conn= pyodbc.connect('DRIVER={ODBC Driver 17 for SQL Server};SERVER='instance+',1433;DATABASE='+database+';Authentication=ActiveDirectoryServicePrincipla;ENCRYPT=yes;TrustedServerCertificate=no;UID'+username+';PWD='+password)

**Comment by gordthompson** [comment, marked as answer]
[sqlserverport](https://pypi.org/project/sqlserverport/) only sends a message on UDP port 1434 if you call `sqlserverport.lookup()`. If you already know the port on which the SQL Server instance is listening then you can just include it in the connection string (as per your example) and UDP port 1434 is not involved.
